arch/arc/core/secureshield/arc_secure.S

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) 2018 Synopsys.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <toolchain.h>
+#include <linker/sections.h>
+#include <arch/cpu.h>
+
+.macro clear_scratch_regs
+	mov r1, 0
+	mov r2, 0
+	mov r3, 0
+	mov r4, 0
+	mov r5, 0
+	mov r6, 0
+	mov r7, 0
+	mov r8, 0
+	mov r9, 0
+	mov r10, 0
+	mov r11, 0
+	mov r12, 0
+.endm
+
+.macro clear_callee_regs
+	mov r25, 0
+	mov r24, 0
+	mov r23, 0
+	mov r22, 0
+	mov r21, 0
+	mov r20, 0
+	mov r19, 0
+	mov r18, 0
+	mov r17, 0
+	mov r16, 0
+
+	mov r15, 0
+	mov r14, 0
+	mov r13, 0
+.endm
+
+GTEXT(arc_go_to_normal)
+GTEXT(_arc_do_secure_call)
+GDATA(arc_s_call_table)
+
+
+SECTION_FUNC(TEXT, _arc_do_secure_call)
+	/* r0-r5: arg1-arg6, r6 is call id */
+	/* the call id should be checked */
+	/* disable normal interrupt happened when processor in secure mode ? */
+	/* seti (0x30 | (ARC_N_IRQ_START_LEVEL-1)) */
+	breq r6, ARC_S_CALL_CLRI, _s_clri
+	breq r6, ARC_S_CALL_SETI, _s_seti
+	push_s blink
+	mov	blink, arc_s_call_table
```

Comment:
Why is a SW dispatch table used instead of using multiple sjli calls and hw dispatching?